tests/unit/core/reducers/test_addons.js

```diff
@@ -553,6 +558,289 @@ describe(__filename, () => {
     });
   });
 
+  describe('updateRatingCounts', () => {
+    function _updateRatingCounts({
+      addonId = 321,
+      oldReview = null,
+      newReview = createInternalReview({ ...fakeReview }),
+    } = {}) {
+      return updateRatingCounts({ addonId, oldReview, newReview });
+    }
+
+    function addonWithRatings(ratings = {}) {
+      return {
+        ...fakeAddon,
+        ratings: {
+          ...fakeAddon.ratings,
+          ...ratings,
+        },
+      };
+    }
+
+    function initStateWithAddon(addon = { ...fakeAddon }) {
+      return addons(undefined, loadAddonResults({ addons: [addon] }));
+    }
+
+    function average(numbers) {
+      return numbers.reduce((total, num) => total + num, 0) / numbers.length;
+    }
+
+    it('does nothing if the add-on has not been loaded', () => {
+      const state = addons(initialState, _updateRatingCounts());
+
+      expect(state).toEqual(initialState);
+    });
+
+    it('increments only the rating count for a newly added score', () => {
+      const ratingCount = 300;
+      const reviewCount = 200;
+      const addon = addonWithRatings({
+        count: ratingCount,
+        text_count: reviewCount,
+      });
+
+      let state = initStateWithAddon(addon);
+      state = addons(
+        state,
+        _updateRatingCounts({
+          addonId: addon.id,
+          oldReview: null,
+          newReview: createInternalReview({
+            ...fakeReview,
+            body: null,
+            score: 5,
+          }),
+        }),
+      );
+
+      const storedAddon = state.byID[addon.id];
```

Comment:
I'd rather use `getAddonByID()` because it should be `${addon.id}` here (in theory). Using the selector will ease refactoring. This comment applies everywhere `state.byID` is used directly.

Reply:
I thought about using `getAddonByID()` but it has a design flaw in that it only accepts `AppState` when it should really accept `AddonsState`. This makes it unusable in the reducer code or the reducer test code without creating an artificial AppState object. I didn't want to do that because it will be fragile, especially in the tests which don't have Flow coverage. Refactoring the selector doesn't seem like a good task for this patch (maybe a future one?).

Reply:
oh heh, then nevermind!

I don't know if it is bad because selectors are also useful to return objects from different sub-states, so passing the `AppState` also makes sense.

Reply:
I've run into this before. I think state selectors should only accept the entire `AppState` if they really need to. Many selectors already operate on their local state type. I filed this to fix it https://github.com/mozilla/addons-frontend/issues/7074